.github/workflows/deploy_to_aws.yml

```diff
@@ -262,73 +267,28 @@ jobs:
-                       -e AWS_DEFAULT_REGION=eu-west-1 \
-                       amazon/aws-cli \
-                       efs describe-file-systems \
-                       --query "FileSystems[?Tags[?Key=='Environment' && Value=='${{ vars.ENV_NAME }}']].FileSystemId" \
-                       --output text)
-          EFS_AP_ID=$(docker run --rm \
-                          -e AWS_ACCESS_KEY_ID=${{ secrets.AWS_ACCESS_KEY_ID }} \
-                          -e AWS_SECRET_ACCESS_KEY=${{ secrets.AWS_SECRET_ACCESS_KEY }} \
-                          -e AWS_DEFAULT_REGION=eu-west-1 \
-                          amazon/aws-cli \
-                          efs describe-access-points \
-                          --query "AccessPoints[?FileSystemId=='$EFS_ID'].AccessPointId" \
-                          --output text)
-          echo "EFS_ID=$EFS_ID" >> $GITHUB_OUTPUT
-          echo "EFS_AP_ID=$EFS_AP_ID" >> $GITHUB_OUTPUT
-          echo "OPENSEARCH_DOMAIN=$OPENSEARCH_DOMAIN" >> $GITHUB_OUTPUT
-      - name: Clear the OpenSearch indexes for ${{ vars.ENV_NAME }}
-        if: ${{ (vars.ENV_NAME != 'Production' && vars.ENV_NAME != 'Staging' && vars.ENV_NAME) && inputs.restore-db == true}}
-        run: |
-          mkdir -p script
-          mkdir -p ssh
-          echo -e "Host *\n\tStrictHostKeyChecking no\n\n" > ssh/config
-          printf "%s\n" "${{ secrets.SSH_PRIVATE_KEY }}" > ssh/id_rsa
-          echo "" >> ssh/id_rsa
-          echo -n ${{ vars.BASTION_IP }} > script/.env
-          cat <<EOF > script/clear_opensearch.sh
-          #!/bin/bash
-          curl -X DELETE https://${{ steps.export_variables.outputs.OPENSEARCH_DOMAIN }}/production-locations --aws-sigv4 "aws:amz:eu-west-1:es" --user "${{ secrets.AWS_ACCESS_KEY_ID }}:${{ secrets.AWS_SECRET_ACCESS_KEY }}"
-          sudo mount -t efs -o tls,accesspoint=${{ steps.export_variables.outputs.EFS_AP_ID }} ${{ steps.export_variables.outputs.EFS_ID }}:/ /mnt
-          sudo rm /mnt/logstash_jdbc_last_run
-          sudo umount /mnt
-          EOF
-          cat <<EOF > script/run.sh
-          chmod 700 /root/.ssh
-          chmod 400 /root/.ssh/id_rsa
-          chmod +x /script/clear_opensearch.sh
-          scp /script/clear_opensearch.sh ec2-user@${{ vars.BASTION_IP }}:
-          ssh ec2-user@${{ vars.BASTION_IP }} ./clear_opensearch.sh
-          EOF
-          docker run --rm  \
-              -v ./script:/script \
-              -v ./ssh:/root/.ssh \
-              kroniak/ssh-client bash /script/run.sh
+              --service=OpenSupplyHub${{vars.ENV_NAME}}AppLogstash
+
+  restore_database:
+    needs: stop_logstash
+    runs-on: self-hosted
+    environment: ${{ inputs.deploy-env || (github.ref_name == 'main' && 'Development') || (startsWith(github.ref_name, 'releases/') && 'Pre-prod') || (startsWith(github.ref_name, 'production-') && 'Production') || (startsWith(github.ref_name, 'sandbox-') && 'Staging') || 'None' }}
+    if:  ${{ inputs.deploy-plan-only == false }}
```

Comment:
Please, add a check for inputs.restore-db == true to ensure this step doesn't start if it's false.

Reply:
We discussed that at the post-deployment stage there may be migrations affecting the data schema and therefore Logstash should not be running at this time. Will there be a situation where the database does not need to be restored, but migrations will still occur?

Reply:
It is the `restore_database` job, which should only be started if `inputs.restore-db` equals `true`. All the steps inside it are triggered only if this condition is true. See the screenshots. I think you can even remove the checks for `inputs.restore-db == true` within the steps if this check is applied at the `restore_database` job level.
1. <img width="1198" alt="Screenshot 2024-09-13 at 11 07 19" src="https://github.com/user-attachments/assets/7b2537e8-5e20-4623-a6d7-2c4e7c30850d">
--------------------
2. <img width="773" alt="Screenshot 2024-09-13 at 11 07 43" src="https://github.com/user-attachments/assets/164bc092-3674-4a38-91d4-fc0eef05c602">

Reply:
@roninzp it looks like it hasn't been done.
I meant to add the check `inputs.restore-db == true` at the `restore_database` job level and remove this check from each step under this job.

Reply:
Why can't this check be done on the job level?

Reply:
if we add this check on job level, all followed jobs be skipped every time when triger not be set to true

Reply:
Got it, thank you!